### PR TITLE
chore: address most PR comments

### PR DESCRIPTION
**Pull request by edwardfoyle**
addresses most PR comments in #4697

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts [hunk 1]
```diff
@@ -194,56 +195,40 @@ class LayerVersionState implements LayerVersionMetadata {
...
+        case Permission.awsOrg:
+          return {
+            type: Permission.awsOrg,
+            orgs: permission.orgs,
+          };
+        case Permission.awsAccounts:
+          return {
+            type: Permission.awsAccounts,
+            accounts: permission.accounts,
+          };
```
Comment (jhockett) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts:217]: nit: return type definitions should be consistent

File: packages/amplify-e2e-tests/src/__tests__/delete.test.ts [hunk 2]
```diff
@@ -171,60 +171,3 @@ async function timeout(timeout: number) {
     setTimeout(resolve, timeout);
   });
 }
-
```
Comment (jhockett) [inline packages/amplify-e2e-tests/src/__tests__/delete.test.ts:174]: Was this deletion intentional?

Comment (edwardfoyle) [reply packages/amplify-e2e-tests/src/__tests__/delete.test.ts:174]: yeah it was removed in master, but I messed up the merge

File: packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts (hunk 1 above)
Comment (edwardfoyle) [reply packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts:217]: These are LayerPermission types and TS is smart enough to figure out the union

**Pull request merged**